test-framework/junit5-mockito/src/main/java/io/quarkus/test/junit/mockito/internal/CreateMockitoSpiesCallback.java

```diff
@@ -18,19 +17,25 @@ public void afterConstruct(Object testInstance) {
             for (Field field : current.getDeclaredFields()) {
                 InjectSpy injectSpyAnnotation = field.getAnnotation(InjectSpy.class);
                 if (injectSpyAnnotation != null) {
-                    Object beanInstance = CreateMockitoMocksCallback.getBeanInstance(testInstance, field, InjectSpy.class);
-                    Object spy = createSpyAndSetTestField(testInstance, field, beanInstance, injectSpyAnnotation.delegate());
-                    MockitoMocksTracker.track(testInstance, spy, beanInstance);
+                    Object contextualReference = CreateMockitoMocksCallback.getContextualReference(testInstance, field,
+                            InjectSpy.class);
+                    Object spy = createSpyAndSetTestField(testInstance, field, contextualReference,
+                            injectSpyAnnotation.delegate());
+                    MockitoMocksTracker.track(testInstance, spy, contextualReference);
                 }
             }
             current = current.getSuperclass();
         }
     }
 
-    private Object createSpyAndSetTestField(Object testInstance, Field field, Object beanInstance, boolean delegate) {
-        Object unwrapped = ClientProxy.unwrap(beanInstance);
-        Object spy = delegate ? Mockito.mock(unwrapped.getClass(), AdditionalAnswers.delegatesTo(unwrapped))
-                : Mockito.spy(unwrapped);
+    private Object createSpyAndSetTestField(Object testInstance, Field field, Object contextualReference, boolean delegate) {
+        Object spy;
+        if (delegate) {
+            spy = Mockito.mock(CreateMockitoMocksCallback.getImplementationClass(contextualReference),
+                    AdditionalAnswers.delegatesTo(contextualReference));
+        } else {
+            spy = Mockito.spy(contextualReference);
```

Comment:
I'm not exactly _sure_, but I _think_ this (after #32405) will make #32300 fail again, at least for normal scoped beans, because we will spy on the client proxy and not on the contextual instance. Did you try that test with these changes?

Reply:
Nope, will do. `ClientProxy.unwrap()` should do the trick...

Reply:
Verified - it failed. Verified - `ClientProxy.unwrap()` works.